cpp/src/quantiles/quantiles.cu

```diff
@@ -157,55 +178,81 @@ namespace{ //unknown
                                                            //(2) for possible types bigger than double, in the future;
                          gdf_context*        ctxt)         //context info
     {
-      return trampoline_exact<T>(col_in, prec, q, t_erased_res, ctxt);
+      return trampoline_exact<T, double>
+                 (col_in, prec, q, t_erased_res, ctxt);
     }
   };
 
   struct trampoline_approx_functor{
+    template <typename UnderLyingT>
+    gdf_error call(gdf_column*  col_in,
+                    double       q,
+                    void*        t_erased_res,
+                    gdf_context* ctxt,
+                    UnderLyingT* data)
+    {
+      return trampoline_exact<UnderLyingT, UnderLyingT>
+                 (col_in, GDF_QUANT_LOWER, q, t_erased_res, ctxt);
+    }
+
     template <typename T>
-    void operator()(gdf_column*  col_in,       //input column;
+    gdf_error operator()(gdf_column*  col_in,       //input column;
                     double       q,            //requested quantile in [0,1]
                     void*        t_erased_res, //type-erased result of same type as column;
                     gdf_context* ctxt)         //context info)
-    {
-      trampoline_approx<T>(col_in, q, t_erased_res, ctxt);
+    { 
+      // ToDo: remove calling "call" function with UnderLying type and directly call "trampoline_exact"
+      // without this, `select_quantile` cannnot be compiled for wrapper types
+      // since double * wrapper types is not defined.
+      // The error occurs at GDF_QUANT_LINEAR and GDF_QUANT_MIDPOINT cases 
+      // even though this call won't get in.
+      T* data = static_cast<T*>( col_in->data );
+      return call(col_in, q, t_erased_res, ctxt, &cudf::detail::unwrap(data[0]));
     }
+
+
   };
 
 }//unknown namespace
 
-gdf_error gdf_quantile_exact(	gdf_column*         col_in,       //input column;
+gdf_error gdf_quantile_exact( gdf_column*         col_in,       //input column;
                               gdf_quantile_method prec,         //precision: type of quantile method calculation
                               double              q,            //requested quantile in [0,1]
-                              void*               t_erased_res, //result; for <exact> should probably be double*; it's void* because
-                                                                //(1) for uniformity of interface with <approx>;
-                                                                //(2) for possible types bigger than double, in the future;
+                              gdf_scalar*         result,       // the result
                               gdf_context*        ctxt)         //context info
 {
   GDF_REQUIRE(!col_in->valid || !col_in->null_count, GDF_VALIDITY_UNSUPPORTED);
   gdf_error ret = GDF_SUCCESS;
   assert( col_in->size > 0 );

```

Comment:
Need to be able to handle empty inputs without erroring.

Reply:
Done.